MovieAppPetProject/MovieAppPetProject/HomeScreen/NewHomeViewController.swift

```diff
@@ -10,46 +10,65 @@ import SDWebImage
 
 class NewHomeViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
 
-    private let nowPlayingViewModel = NowPlayingViewModel()
+    // MARK: - IBOutlets
+    @IBOutlet private weak var newTableView: UITableView!
 
-    @IBOutlet weak var newTableView: UITableView!
+    // MARK: - Variables
+    private let nowPlayingViewModel = NowPlayingViewModel()
 
+    // MARK: - Functions
     override func viewDidLoad() {
         super.viewDidLoad()
         fetchNowPlayingMovies()
     }
 
-    func fetchNowPlayingMovies() {
-        nowPlayingViewModel.fetchNowPlayingMovies { [weak self] result in
-            switch result {
-            case .success:
-                DispatchQueue.main.async {
-                    self?.newTableView.reloadData()
-                }
-            case .failure(let error):
-                print("Error fetching now playing movies: \(error)")
-            }
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard segue.identifier == Constants.Identifiers.goToMovieDetails,
+              let movieDetailsViewController = segue.destination as? MovieDetailsViewController,
+              let selectedNowPlayingMovie = sender as? NowPlayingResults else {
+            return
         }
+
+        let viewModel = MovieDetailsViewModel()
```

Comment:
this is not the best. This newHomeScreenViewController should not be speaking to MovieDetailsViewModel and NowPlayingViewModel

Reply:
will refactor accordingly